modifiers/caliper/modifier.py

```diff
@@ -30,58 +19,88 @@ class Caliper(BasicModifier):
 
     maintainers("pearce8")
 
-    _cali_datafile = "{experiment_run_dir}/{experiment_name}.cali"
-
-    _default_mode = "time"
-
-    add_mode(
-        mode_name=_default_mode,
-        mode_option="time.exclusive",
-        description="Platform-independent collection of time (default mode)",
-    )
-
-    env_var_modification(
-        "CALI_CONFIG",
-        "spot(output={}{})".format(_cali_datafile, "${CALI_CONFIG_MODE}"),
-        method="set",
-        modes=[_default_mode],
-    )
-
-    add_mode(
-        mode_name="mpi",
-        mode_option="profile.mpi",
-        description="Profile MPI functions",
-    )
-
-    add_mode(
-        mode_name="cuda",
-        mode_option="profile.cuda",
-        description="Profile CUDA API functions",
-    )
-
-    add_mode(
-        mode_name="topdown-counters-all",
-        mode_option="topdown-counters.all",
-        description="Raw counter values for Intel top-down analysis (all levels)",
-    )
-
-    add_mode(
-        mode_name="topdown-counters-toplevel",
-        mode_option="topdown-counters.toplevel",
-        description="Raw counter values for Intel top-down analysis (top level)",
-    )
-
-    add_mode(
-        mode_name="topdown-all",
-        mode_option="topdown.all",
-        description="Top-down analysis for Intel CPUs (all levels)",
-    )
-
-    add_mode(
-        mode_name="topdown-toplevel",
-        mode_option="topdown.toplevel",
-        description="Top-down analysis for Intel CPUs (top level)",
-    )
+    def determine_cali_config(self):
+        if self.app.name != "raja-perf":
+            self.env_var_modification(
+                "CALI_CONFIG",
+                "spot(output={}{})".format(_cali_datafile, "${CALI_CONFIG_MODE}"),
+                method="set",
+                modes=[_default_mode],
+            )
+        else:
+            pass
+
+    def inherit_from_application(self, app):
+        super().inherit_from_application(app)
+        self.app = app
+
+        self.determine_cali_config()
+
+        self.add_modes()
+
+    def add_mode(self, mode_name, mode_option, description, app):
+        self.mode(
+            name=mode_name,
+            description=description,
+        )
+
+        self.env_var_modification(
+            "CALI_CONFIG_MODE",
+            mode_option,
+            method="append",
+            separator="," if app.name != "raja-perf" else "",
```

Comment:
@michaelmckinsey1 Can we have multiple CALI_CONFIG_MODEs defined for raja-perf?

Reply:
Yes, I see the problem. [52a1368](https://github.com/LLNL/benchpark/pull/603/commits/52a1368165f82cc19ebf6117c221567f9fbc6be3) fixes this.